public/app/features/search/service/bluge.ts

```diff
@@ -141,7 +141,7 @@ async function doSearchQuery(query: SearchQuery): Promise<QueryResponse> {
                 search: {
                   ...(target?.search ?? {}),
                   from,
-                  limit: Math.max(limit, nextPageSizes),
+                  limit: Math.min(limit, nextPageSizes),
```

Comment:
this *may* be the issue you were seeing @ArturWierzbicki, with bigger blocks it should not really matter though